parachains/runtimes/assets/asset-hub-polkadot/tests/tests.rs

```diff
@@ -646,3 +651,131 @@ asset_test_utils::include_create_and_manage_foreign_assets_for_local_consensus_p
 		assert_eq!(ForeignAssets::asset_ids().collect::<Vec<_>>().len(), 1);
 	})
 );
+
+fn bridging_to_asset_hub_kusama() -> asset_test_utils::test_cases_over_bridge::TestBridgingConfig {
+	asset_test_utils::test_cases_over_bridge::TestBridgingConfig {
+		bridged_network: bridging::KusamaNetwork::get(),
+		local_bridge_hub_para_id: bridging::BridgeHubPolkadotParaId::get(),
+		local_bridge_hub_location: bridging::BridgeHubPolkadot::get(),
+		bridged_target_location: bridging::AssetHubKusama::get(),
+	}
+}
+
+#[test]
+fn limited_reserve_transfer_assets_for_native_asset_over_bridge_works() {
+	asset_test_utils::test_cases_over_bridge::limited_reserve_transfer_assets_for_native_asset_works::<
+		Runtime,
+		XcmConfig,
+		ParachainSystem,
+		XcmpQueue,
+		LocationToAccountId,
+	>(
+		collator_session_keys(),
+		ExistentialDeposit::get(),
+		AccountId::from(ALICE),
+		Box::new(|runtime_event_encoded: Vec<u8>| {
+			match RuntimeEvent::decode(&mut &runtime_event_encoded[..]) {
+				Ok(RuntimeEvent::PolkadotXcm(event)) => Some(event),
+				_ => None,
+			}
+		}),
+		Box::new(|runtime_event_encoded: Vec<u8>| {
+			match RuntimeEvent::decode(&mut &runtime_event_encoded[..]) {
+				Ok(RuntimeEvent::XcmpQueue(event)) => Some(event),
+				_ => None,
+			}
+		}),
+		bridging_to_asset_hub_kusama,
+		WeightLimit::Unlimited,
+	)
+}
+
+#[test]
+fn receive_reserve_asset_deposited_ksm_from_asset_hub_kusama_works() {
+	const BLOCK_AUTHOR_ACCOUNT: [u8; 32] = [13; 32];
+	asset_test_utils::test_cases_over_bridge::receive_reserve_asset_deposited_from_different_consensus_works::<
+		Runtime,
+		XcmConfig,
+		LocationToAccountId,
+		ForeignAssetsInstance,
+	>(
+		collator_session_keys().add(collator_session_key(BLOCK_AUTHOR_ACCOUNT)),
+		ExistentialDeposit::get(),
+		AccountId::from([73; 32]),
+		AccountId::from(BLOCK_AUTHOR_ACCOUNT),
+		// receiving DOTs
+		(MultiLocation { parents: 2, interior: X1(GlobalConsensus(Kusama)) }, 1000000000000, 1_000_000_000),
+		bridging_to_asset_hub_kusama,
+		(X1(PalletInstance(53)), GlobalConsensus(Kusama), X1(Parachain(1000)))
+	)
+}
+
+/// Tests expected configuration of isolated `pallet_xcm::Config::XcmReserveTransferFilter`.
+#[test]
+fn xcm_reserve_transfer_filter_works() {
+	// prepare assets
+	let only_native_assets = || vec![MultiAsset::from((DotLocation::get(), 1000))];
+	let only_trust_backed_assets = || {
+		vec![MultiAsset::from((
+			AssetIdForTrustBackedAssetsConvert::convert_back(&12345).unwrap(),
+			2000,
+		))]
+	};
+	let only_sibling_foreign_assets =
+		|| vec![MultiAsset::from((MultiLocation::new(1, X1(Parachain(12345))), 3000))];
+	let only_different_global_consensus_foreign_assets = || {
+		vec![MultiAsset::from((
+			MultiLocation::new(2, X2(GlobalConsensus(Kusama), Parachain(12345))),
+			4000,
+		))]
+	};
+
+	// prepare destinations
+	let relaychain = MultiLocation::parent();
+	let sibling_parachain = MultiLocation::new(1, X1(Parachain(1000)));
```

Comment:
Does this test execute in a different context than Asset Hub? Because this would be `Here`, not a sibling.

Reply:
ah, thank you, good catch, I better changed 1000 to 54321 para_id to avoid confusion,

anyway, `pallet_xcm::Config::XcmReserveTransferFilter` does not work with context, it just matches origin/assets
```
let value = (origin_location, assets.into_inner());
ensure!(T::XcmReserveTransferFilter::contains(&value), Error::<T>::Filtered);
```